### PR TITLE
bitwarden: 2023.10.1 -> 2023.12.0

### LINKED ISSUE #273283: Build failure: bitwarden
### Steps To Reproduce

1. build bitwarden

### Build log

```
error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'shell'
         whose name attribute is located at /nix/store/xmi0h8i4zqbhnswlhbsl6ikvdacaydqp-nixpkgs/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:348:7

       … while evaluating attribute 'buildInputs' of derivation 'shell'

         at /nix/store/xmi0h8i4zqbhnswlhbsl6ikvdacaydqp-nixpkgs/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:395:7:

          394|       depsHostHost                = elemAt (elemAt dependencies 1) 0;
          395|       buildInputs                 = elemAt (elemAt dependencies 1) 1;
             |       ^
          396|       depsTargetTarget            = elemAt (elemAt dependencies 2) 0;

       … while evaluating derivation 'bitwarden-2023.10.1'
         whose name attribute is located at /nix/store/xmi0h8i4zqbhnswlhbsl6ikvdacaydqp-nixpkgs/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:348:7

       … while evaluating attribute 'installPhase' of derivation 'bitwarden-2023.10.1'

         at /nix/store/xmi0h8i4zqbhnswlhbsl6ikvdacaydqp-nixpkgs/nixpkgs/pkgs/tools/security/bitwarden/default.nix:132:3:

          131|
          132|   installPhase = ''
             |   ^
          133|     runHook preInstall

       … from call site

         at /nix/store/xmi0h8i4zqbhnswlhbsl6ikvdacaydqp-nixpkgs/nixpkgs/pkgs/stdenv/generic/check-meta.nix:487:16:

          486|         {
          487|           no = handleEvalIssue { inherit meta attrs; } { inherit (validity) reason errormsg; };
             |                ^
          488|           warn = handleEvalWarning { inherit meta attrs; } { inherit (validity) reason errormsg; };

       … while calling 'handleEvalIssue'

         at /nix/store/xmi0h8i4zqbhnswlhbsl6ikvdacaydqp-nixpkgs/nixpkgs/pkgs/stdenv/generic/check-meta.nix:250:38:

          249|
          250|   handleEvalIssue = { meta, attrs }: { reason , errormsg ? "" }:
             |                                      ^
          251|     let

       error: Package ‘electron-25.9.0’ in /nix/store/xmi0h8i4zqbhnswlhbsl6ikvdacaydqp-nixpkgs/nixpkgs/pkgs/development/tools/electron/binary/generic.nix:35 is marked as insecure, refusing to evaluate.


       Known issues:
        - Electron version 25.9.0 is EOL

       You can install it anyway by allowing this package, using the
       following methods:

       a) To temporarily allow all insecure packages, you can use an environment
          variable for a single invocation of the nix tools:

            $ export NIXPKGS_ALLOW_INSECURE=1

          Note: When using `nix shell`, `nix build`, `nix develop`, etc with a flake,
                then pass `--impure` in order to allow use of environment variables.

       b) for `nixos-rebuild` you can add ‘electron-25.9.0’ to
          `nixpkgs.config.permittedInsecurePackages` in the configuration.nix,
          like so:

            {
              nixpkgs.config.permittedInsecurePackages = [
                "electron-25.9.0"
              ];
            }

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
          ‘electron-25.9.0’ to `permittedInsecurePackages` in
          ~/.config/nixpkgs/config.nix, like so:

            {
              permittedInsecurePackages = [
                "electron-25.9.0"
              ];
            }

```

### Notify maintainers
@amarshall 
@Kiwi 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.64, NixOS, 24.05 (Uakari), 24.05pre554642.f5c27c6136db`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(kunny): `"home-manager, nixpkgs"`
 - nixpkgs: `/home/kunny/.nix-defexpr/channels/nixpkgs`

```

---

### PR DESCRIPTION
**Pull request by dotlambda**
## Description of changes
Diff: https://github.com/bitwarden/clients/compare/desktop-v2023.10.1...desktop-v2023.12.0

Changelog: https://github.com/bitwarden/clients/releases/tag/desktop-v2023.12.0

closes https://github.com/NixOS/nixpkgs/issues/273283
closes https://github.com/NixOS/nixpkgs/pull/272658

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by amarshall — APPROVED** [review]
Verified build and basic functionality of desktop app.

Verified build and `bw --help` for CLI.

I did build by cherry-picking these commits onto nixos-unstable (2c7f3c0fb7c08a0814627611d9d7d45ab6d75335) branch since didn’t feel like waiting for electron to build.

**Review by SuperSandro2000 — APPROVED** [review]
Didn't test on my machine but looks good to me

File: pkgs/tools/security/bitwarden/default.nix [hunk 1]
```diff
@@ -82,12 +82,14 @@ in buildNpmPackage rec {
     libsecret
   ];
 
-  preBuild = ''
+  # FIXME add back once upstream moves to Electron >= 26
+  # we use electron_26 because electron_25 is EOL
+  /*preBuild = ''
```
[the remark this answers is not in the document]
Comment (dotlambda) [reply pkgs/tools/security/bitwarden/default.nix:87]: Would a comment behind the `version` attribute be sufficient?

**Comment by SuperSandro2000** [comment]
I am not sure if the following errors are related to using electron 26

```
01:28:13.617 › TypeError: Cannot read properties of undefined (reading 'deleteBiometricKey')
    at BiometricsService.<anonymous> (/nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:48310:55)
    at Generator.next (<anonymous>)
    at /nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:48218:71
    at new Promise (<anonymous>)
    at biometrics_service_awaiter (/nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:48214:12)
    at BiometricsService.deleteBiometricKey (/nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:48308:16)
    at DesktopCredentialStorageListener.<anonymous> (/nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:48517:45)
    at Generator.next (<anonymous>)
    at /nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:48394:71
    at new Promise (<anonymous>)
01:28:13.618 › TypeError: Cannot read properties of undefined (reading 'deleteBiometricKey')
    at BiometricsService.<anonymous> (/nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:48310:55)
    at Generator.next (<anonymous>)
    at /nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:48218:71
    at new Promise (<anonymous>)
    at biometrics_service_awaiter (/nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:48214:12)
    at BiometricsService.deleteBiometricKey (/nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:48308:16)
    at DesktopCredentialStorageListener.<anonymous> (/nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:48517:45)
    at Generator.next (<anonymous>)
    at /nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:48394:71
    at new Promise (<anonymous>)
(node:893169) UnhandledPromiseRejectionWarning: TypeError: Cannot read properties of undefined (reading 'setContextMenu')
    at TrayMain.updateContextMenu (/nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:47357:23)
    at MessagingMain.updateTrayMenu (/nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:46857:28)
    at MessagingMain.onMessage (/nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:46783:22)
    at MessagingMain.<anonymous> (/nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:46774:151)
    at Generator.next (<anonymous>)
    at /nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:46753:71
    at new Promise (<anonymous>)
    at messaging_main_awaiter (/nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:46749:12)
    at IpcMainImpl.<anonymous> (/nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:46774:79)
    at IpcMainImpl.emit (node:events:513:28)
(node:893169) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 22)
```

and when exiting:

```
01:28:25.591 › TypeError: Object has been destroyed
    at WindowMain.<anonymous> (/nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:47777:36)
    at Generator.next (<anonymous>)
    at /nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:47547:71
    at new Promise (<anonymous>)
    at window_main_awaiter (/nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:47543:12)
    at WindowMain.updateWindowState (/nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:47772:16)
    at WindowMain.<anonymous> (/nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:47706:28)
    at Generator.next (<anonymous>)
    at /nix/store/n501n5zszk113sq0wiyv47cxlqavhqgp-bitwarden-2023.12.0/opt/Bitwarden/resources/app.asar/main.js:47547:71
    at new Promise (<anonymous>)
Segmentation fault (core dumped)
```

**Comment by dotlambda** [comment]
> TypeError: Cannot read properties of undefined (reading 'deleteBiometricKey')

Reading through the `bitwarden/clients` code this simply doesn't seem to be implemented for Linux.

**Review by amarshall — APPROVED** [review]

**Review by patka-123 — APPROVED** [review]
Result of `nixpkgs-review pr 272541` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>bitwarden</li>
    <li>bitwarden-cli</li>
  </ul>
</details>

<hr>

Looks good to me. Also tested some things manually and it seems to work well.

**Pull request merged**